nano/core_test/network.cpp

```diff
@@ -74,48 +121,6 @@ TEST (network, construction_without_specified_port)
 	EXPECT_EQ (port, node->bootstrap.endpoint ().port ());
 }
 
-TEST (network, self_discard)
```

Comment:
Why was this test removed? The concept of self-connecting exists in TCP too.

Reply:
This seems to be purely UDP. As we have not been testing TCP for this, I think the best approach is to create an issue for us to create a test for it as a further step. Thinking about to do the same with the other tests I removed. The criteria used to remove them as that they were either not testing anything or that seemed to be UDP exclusive.

Reply:
Let's take the opposite approach, we do not remove tests unless we are sure we do not need them.
so let's reinstate this test as it was, using UDP, and remove or convert in a different PR.